File: gpao-tex-ui/source/main/src/app/services/product/models/produit.ts
/* tslint:disable */
/* eslint-disable */
import { Couleur } from '../models/couleur';
import { ElementTraitement } from '../models/element-traitement';
import { LigneProduit } from '../models/ligne-produit';
import { MesureProduit } from '../models/mesure-produit';
import { PilotageEfficience } from '../models/pilotage-efficience';
import { ProduitGoShippement } from '../models/produit-go-shippement';
import { Site } from '../models/site';
import { SousFamilleProduit } from '../models/sous-famille-produit';
import { Taille } from '../models/taille';
export interface Produit {
  archived?: boolean;
  cadence?: number;
  codeConception?: string;
  couleurs?: Array<Couleur>;
  creationDate?: string;
  creatorIdentifier?: string;
  deletionDate?: string;
  designation?: string;
  elementTraitements?: Array<ElementTraitement>;
  etat?: string;
  ficheBesoin?: boolean;
  id?: number;
  latestUpdaterId?: string;
  ligneProduit?: LigneProduit;
  mesureProduits?: Array<MesureProduit>;
  modificationDate?: string;
  partieInteresseId?: number;
  pilotageEfficiences?: Array<PilotageEfficience>;
  prixMajore?: number;
  prixUnitaire?: number;
  produitGoShippements?: Array<ProduitGoShippement>;
  quantite?: number;
  reference?: string;
  sansCouleur?: boolean;
  sansTaille?: boolean;
  site?: Site;
  sousFamilleProduit?: SousFamilleProduit;
  tailles?: Array<Taille>;
  version?: number;
}

File: gpao-tex-core/produit/src/main/java/com/gpro/consulting/produit/model/Produit.java
package com.gpro.consulting.produit.model;

import com.gpro.consulting.produit.dto.PartieInteresseValue;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Table(name = "eb_produit")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Produit extends BaseEntity {
    @Id
    @GeneratedValue(generator = "PRODUIT_ID_GENERATOR", strategy = GenerationType.SEQUENCE)
    @SequenceGenerator(name = "PRODUIT_SEQ", sequenceName = "PRODUIT_SEQ", allocationSize = 1)
    @Column(name = "id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "eb_sfprod_id")
    private SousFamilleProduit sousFamilleProduit;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "com_site_id")
    private Site site;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "theme_id")
    private Theme theme;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ligneProduit_id")
    private LigneProduit ligneProduit;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "produit", cascade = CascadeType.REMOVE)
    private List<ProduitGoShippement> produitGoShippements;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "produit", cascade = CascadeType.REMOVE)
    private List<MesureProduit> mesureProduits;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "produit", cascade = CascadeType.REMOVE)
    private List<Taille> tailles;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "produit", cascade = CascadeType.REMOVE)
    private List<Couleur> couleurs;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "produit", cascade = CascadeType.REMOVE)
    private List<PilotageEfficience> pilotageEfficiences;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "produit", cascade = CascadeType.REMOVE)
    private List<ElementTraitement> elementTraitements;

    @Transient
    private PartieInteresseValue partieInteresse;

    @Column(name = "pi_pi_id")
    private Long partieInteresseId;
    private String reference;
    private String designation;
    @Column(name = "fiche_besoin")
    private boolean ficheBesoin = false;
    @Column(name = "sans_taille")
    private boolean sansTaille = false;
    @Column(name = "sans_couleur")
    private boolean sansCouleur = false;
    @Column(name = "code_conception")
    private String codeConception;
    private Integer quantite;
    private Integer cadence;
    @Column(name = "prix_unitaire")
    private BigDecimal prixUnitaire;
    @Column(name = "prix_majore")
    private BigDecimal prixMajore;
    private String etat;
}
